packages/entities/entities-plugins/src/components/free-form/Datakit/flow-editor/store/history.ts

```diff
@@ -52,6 +52,10 @@ export function useTaggedHistory<T>(
     baseCommit()
     if (undoStack.value.length > 1) {
       undoStack.value.splice(1, 1)
+    } else if (tag === '*' && undoStack.value.length === 1) {
+      // If `tag` is "*" and only one history checkpoint is present,
+      // remove it as we don't have other checkpoints to collapse into.
+      undoStack.value.splice(0, 1)
```

Comment:
Does this mean committing with `*` is the first commit and this should be considered as a part of the initial state, right? I think we can improve the comment to make it clearer.